apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs

```diff
@@ -276,5 +278,190 @@ private static string ValidateTableName(string table, string parameterName)
+                        if (c == '\n')
+                        {
+                            isInSingleLineComment = false;
+                            // Include the line feed in the result.
+                            res.Append(c);
+                        }
+                    }
+                    else if (isInMultiLineComment)
+                    {
+                        if (sql.Length > index + 1 && c == s_asterisk && sql[index + 1] == s_slash)
+                        {
+                            isInMultiLineComment = false;
+                            index++;
+                        }
+                    }
+                    else
+                    {
+                        if (c == s_dash
+                            || (sql.Length > index + 1 && c == s_hyphen && sql[index + 1] == s_hyphen))
+                        {
+                            // This is a single line comment.
+                            isInSingleLineComment = true;
+                        }
+                        else if (sql.Length > index + 1 && c == s_slash && sql[index + 1] == s_asterisk)
+                        {
+                            isInMultiLineComment = true;
+                            index++;
+                        }
+                        else
+                        {
+                            if (c == s_singleQuote || c == s_doubleQuote || c == s_backtickQuote)
+                            {
+                                isInQuoted = true;
+                                startQuote = c;
+                                // Check whether it is a triple-quote.
+                                if (sql.Length > index + 2
+                                    && sql[index + 1] == startQuote
+                                    && sql[index + 2] == startQuote)
+                                {
+                                    isTripleQuoted = true;
+                                    res.Append(c).Append(c);
+                                    index += 2;
+                                }
+                            }
+                            res.Append(c);
+                        }
+                    }
+                }
+                index++;
+            }
+            if (isInQuoted)
+            {
+                throw new SpannerException(
+                    ErrorCode.InvalidArgument, $"SQL statement contains an unclosed literal: {sql}");
+            }
+            if (res.Length > 0 && res[res.Length - 1] == ';')
+            {
+                res.Remove(res.Length - 1, 1);
+            }
+            return res.ToString();
+        }
+
+        /// <summary>
+        /// First removes all comments from the entire sql string, and then removes any statement hints
+        /// at the start of the sql statement. The returned statement should only be used to determine the
+        /// type of statement. The original statement should always be passed on to Cloud Spanner.
```

Comment:
This is what is not happening in all cases because of line 262.

Reply:
See also the note above: Those commands are converted to mutations, so the command text is never sent to Spanner.